### DISCUSSION TITLE
How to set up a rolling window partition dependency between assets?

### DISCUSSION THREAD

**Discussion by sryza** [dagster-io/dagster#14636] [Q&A] [answered]
I want each daily partition of my downstream asset to depend on three partitions of the upstream asset:
- The partition for the same day
- The partition for the prior two days

**Comment by sryza** [comment, marked as answer]
You can accomplish this with the `start_offset` argument on `TimeWindowPartitionMapping`:

```python
from dagster import AssetIn, DailyPartitionsDefinition, TimeWindowPartitionMapping, asset


@asset(partitions_def=DailyPartitionsDefinition(start_date="2020-01-01"))
def asset1():
    ...


@asset(
    partitions_def=DailyPartitionsDefinition(start_date="2020-01-01"),
    ins={"asset1": AssetIn(partition_mapping=TimeWindowPartitionMapping(start_offset=-2))},
)
def asset2(asset1):
    ...
```

**Comment by abhinavDhulipala** [comment]
What happens to the data on expiry? Does it, along with the runs associated logs/events get garbage collected upon being orphaned? Also, is it possible to put this in the partition definition itself? I.e have asset1 be an asset where the data has a static TTL and all derived data is subsequently destroyed as well?
